fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/QuantifiedApply2JoinRule.java

```diff
@@ -49,8 +49,10 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
         // NOT IN to ANTI-JOIN or NULL_AWARE_LEFT_ANTI_JOIN
         OptExpression joinExpression;
         if (ipo.isNotIn()) {
-            //@TODO: if will can filter null, use left-anti-join
-            joinExpression = new OptExpression(new LogicalJoinOperator(JoinOperator.NULL_AWARE_LEFT_ANTI_JOIN,
+            boolean operandsNullable = bpo.getChildren().stream().anyMatch(ScalarOperator::isNullable);
+            JoinOperator joinType = operandsNullable ?
```

Comment:
The nullable in join node is not trusted

Reply:
oh...bad news.